### PR TITLE
support controller manager prefix and namespace for multiple controller managers

### PR DESCRIPTION
**Pull request by pabloinigoblasco**
This PR is created due to the need of having multiple controller managers to control multiple arms.

### CODE REVIEW

**Review by ahcorde — CHANGES_REQUESTED** [review]
Did you try to use `<plugin><ros><namespace></namespace></ros></plugin>` ?

```
<gazebo>
    <plugin filename="libgazebo_ros2_control.so" name="gazebo_ros2_control">
      <robot_param>...</robot_param>
      <robot_param_node>...</robot_param_node>
      <parameters>....</parameters>
      <ros>
         <namespace></namespace>
      </ros>
    </plugin>
</gazebo>

File: gazebo_ros2_control/src/gazebo_ros2_control_plugin.cpp [hunk 1]
```diff
@@ -196,6 +196,18 @@ void GazeboRosControlPlugin::Load(gazebo::physics::ModelPtr parent, sdf::Element
...
+  std::string controllerManagerNodeName{"controller_manager"};
+
+  std::string controllerManagerPrefixNodeName =
+    sdf->Get<std::string>("controller_manager_prefix_node_name");
+  if (!controllerManagerPrefixNodeName.empty()) {
+    controllerManagerNodeName = controllerManagerPrefixNodeName + "_" + controllerManagerNodeName;
+  }
+
+    std::string controllerManagerNodeNameNamespace{""};
+    controllerManagerNodeNameNamespace = sdf->Get<std::string>("controller_manager_namespace");
```
Comment (bmagyar) [inline gazebo_ros2_control/src/gazebo_ros2_control_plugin.cpp:208]:
```suggestion
  std::string controllerManagerNodeNameNamespace{""};
  controllerManagerNodeNameNamespace = sdf->Get<std::string>("controller_manager_namespace");
```
we should probably set up a clang-format file here too

File: gazebo_ros2_control/src/gazebo_system.cpp [hunk 2]
```diff
@@ -27,6 +27,8 @@
 #include "hardware_interface/hardware_info.hpp"
 #include "hardware_interface/types/hardware_interface_type_values.hpp"
 
+using CallbackReturn = rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn;
```
Comment (bmagyar) [inline gazebo_ros2_control/src/gazebo_system.cpp:30]:
```suggestion
using CallbackReturn = rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn;
```


is this needed?

**Review by bmagyar — CHANGES_REQUESTED** [review]

File: gazebo_ros2_control/src/gazebo_system.cpp (hunk 2 above)
Comment (destogl) [reply gazebo_ros2_control/src/gazebo_system.cpp:30]: It should be needed, but probably under "namespace" scope and not global scope.

**Comment by destogl** [comment]
Replaced by #147

**Pull request closed without merging**